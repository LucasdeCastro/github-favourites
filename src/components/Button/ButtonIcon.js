import React from "react";
import * as FontAwesome from "react-icons/lib/fa";

export default ({ onClick, text, iconName }) => {
  const Icon = FontAwesome[iconName];
  return (
    <button onClick={onClick}>
      <Icon /> {text || ""}
    </button>
  );
};
