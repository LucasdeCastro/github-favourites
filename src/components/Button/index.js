import Icon from "./ButtonIcon";

export default {
  Icon
};
