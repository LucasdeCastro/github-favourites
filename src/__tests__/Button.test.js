import React from "react";
import renderer from "react-test-renderer";
import Button from "../components/Button/ButtonIcon";

describe("[ButtonIcon]", () => {
  test("try render component", () => {
    const component = renderer.create(
      <Button iconName="FaClose" text={"close"} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("try render component without iconName", () => {
    const component = renderer.create(<Button text={"close"} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("try render component without text", () => {
    const component = renderer.create(<Button iconName={"FaClose"} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
