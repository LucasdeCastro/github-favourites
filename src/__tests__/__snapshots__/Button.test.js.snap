// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ButtonIcon] try render component 1`] = `
<button
  onClick={undefined}
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m33.5 29.5q0 0.9-0.7 1.5l-3 3.1q-0.6 0.6-1.5 0.6t-1.5-0.6l-6.6-6.6-6.5 6.6q-0.7 0.6-1.6 0.6t-1.5-0.6l-3-3.1q-0.6-0.6-0.6-1.5t0.6-1.5l6.5-6.6-6.5-6.5q-0.6-0.7-0.6-1.6t0.6-1.5l3-3q0.6-0.6 1.5-0.6t1.6 0.6l6.5 6.6 6.6-6.6q0.6-0.6 1.5-0.6t1.5 0.6l3.1 3q0.6 0.7 0.6 1.5t-0.6 1.6l-6.6 6.5 6.6 6.6q0.6 0.6 0.6 1.5z"
      />
    </g>
  </svg>
   
  close
</button>
`;

exports[`[ButtonIcon] try render component without iconName 1`] = `null`;

exports[`[ButtonIcon] try render component without text 1`] = `
<button
  onClick={undefined}
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m33.5 29.5q0 0.9-0.7 1.5l-3 3.1q-0.6 0.6-1.5 0.6t-1.5-0.6l-6.6-6.6-6.5 6.6q-0.7 0.6-1.6 0.6t-1.5-0.6l-3-3.1q-0.6-0.6-0.6-1.5t0.6-1.5l6.5-6.6-6.5-6.5q-0.6-0.7-0.6-1.6t0.6-1.5l3-3q0.6-0.6 1.5-0.6t1.6 0.6l6.5 6.6 6.6-6.6q0.6-0.6 1.5-0.6t1.5 0.6l3.1 3q0.6 0.7 0.6 1.5t-0.6 1.6l-6.6 6.5 6.6 6.6q0.6 0.6 0.6 1.5z"
      />
    </g>
  </svg>
   
  
</button>
`;
